drivers/sensor/tmag5273/tmag5273.c

```diff
@@ -0,0 +1,1218 @@
+		return -EIO;
+	}
+
+	if (regdata != TMAG5273_MANUFACTURER_ID_LSB) {
+		LOG_DBG("uncexpected manufacturer id");
+		return -EINVAL;
+	}
+
+	retval = i2c_reg_read_byte_dt(&drv_cfg->i2c,
+		TMAG5273_REG_MANUFACTURER_ID_MSB,
+		&regdata);
+
+	if (retval < 0) {
+		LOG_DBG("Could not read MSB of manufacturer id |%d|", retval);
+		return -EIO;
+	}
+
+	if (regdata != TMAG5273_MANUFACTURER_ID_MSB) {
+		LOG_DBG("uncexpected manufacturer id");
+		return -EINVAL;
+	}
+
+	/* check device id*/
+	retval = i2c_reg_read_byte_dt(&drv_cfg->i2c,
+		TMAG5273_REG_DEVICE_ID,
+		&regdata);
+
+	if (retval < 0) {
+		LOG_DBG("Could not read device id |%d|", retval);
+		return -EIO;
+	}
+
+	tmag5273_read_device_status(dev);
+
+	retval = tmag5273_reset_device_status(dev);
+
+	if (retval < 0) {
+		LOG_DBG("Could not reset DEVICE_STATUS register |%d|", retval);
+		return -EIO;
+	}
+
+	drv_data->version = TMAG5273_GET_REG_BITS(regdata, TMAG5273_VER);
+
+	/* magnetic measurement range based on version, apply correct one */
+	if (drv_data->version == TMAG5273_VER_TMAG5273X1) {
+		drv_data->xyz_range = (drv_cfg->meas_range != TMAG5273_AXIS_RANGE_LOW)
+			? TMAG5273_MEAS_RANGE_HIGH_MT_VER1 : TMAG5273_MEAS_RANGE_LOW_MT_VER1;
+	} else if (drv_data->version == TMAG5273_VER_TMAG5273X2) {
+		drv_data->xyz_range = (drv_cfg->meas_range != TMAG5273_AXIS_RANGE_LOW)
+			? TMAG5273_MEAS_RANGE_HIGH_MT_VER2 : TMAG5273_MEAS_RANGE_LOW_MT_VER2;
+	} else {
+		LOG_DBG("unexpected version");
+		return -EINVAL;
+	}
+
+	/* deactivate interrupts if necessary (will be overwritten if needed) */
+	regdata = (TMAG5273_INT_MODE_NONE << TMAG5273_INT_MODE_POS)
+		| (drv_cfg->mask_intb_value << TMAG5273_INT_MASK_INTB_POS);
+
+	retval = i2c_reg_write_byte_dt(&drv_cfg->i2c,
+			TMAG5273_REG_INT_CONFIG_1,
+			regdata);
+
+	if (retval < 0) {
+		LOG_DBG("Error deactivating interrupts");
+		return -EIO;
+	}
+
+	/* set settings */
+	retval = tmag5273_init_sensor_settings(drv_cfg);
+
+	if (retval < 0) {
+		LOG_DBG("error setting sensor configuration |%d|", retval);
+		return retval;
+	}
+
+	retval = tmag5273_init_device_config(drv_cfg);
+
+	if (retval < 0) {
+		LOG_DBG("error setting device configuration |%d|", retval);
+		return retval;
+	}
+
+	return 0;
+}
+
+/* ----------------------------------------------------------- */
+
+static const struct sensor_driver_api tmag5273_driver_api = {
+	.attr_set = tmag5273_attr_set,
+	.attr_get = tmag5273_attr_get,
+	.sample_fetch = tmag5273_sample_fetch,
+	.channel_get = tmag5273_channel_get,
+};
+
+/* ----------------------------------------------------------- */
+
+/** Instantiation macro */
+#define TMAG5273_DEFINE(inst) \
+	static const struct tmag5273_config tmag5273_driver_cfg##inst = { \
+		.i2c = I2C_DT_SPEC_INST_GET(inst), \
+		.x_axis = (bool)(DT_INST_PROP(inst, axis) & TMAG5273_AXIS_X), \
+		.y_axis = (bool)(DT_INST_PROP(inst, axis) & TMAG5273_AXIS_Y), \
+		.z_axis = (bool)(DT_INST_PROP(inst, axis) & TMAG5273_AXIS_Z), \
```

Comment:
Would it be simpler to have a separate boolean devicetree property for each axis?

Reply:
Granted. It's the same cause as for you comment about the justification in the bindings-file: I assumed that setting the XYZ-measurement to "default on" does make the most sense. If this shouldn't be the case, I'll can change this accordingly.

Reply:
(also provided dt-bindings define for a better representation)